apis/apps/v1alpha1/apimanager_types.go

```diff
@@ -505,6 +505,7 @@ type DeprecatedSystemS3Spec struct {
 
 type SystemS3Spec struct {
 	ConfigurationSecretRef v1.LocalObjectReference `json:"configurationSecretRef"`
+	S3STSEnabled           bool                    `json:"s3StsEnabledSts,omitempty"`
```

Comment:
The type of secret is not determined by the APIManager CR, but from the fields in the secret. That is what we decided in https://issues.redhat.com/browse/THREESCALE-8772

Reply:
after offline discussion, we changed our  mind.  There should be a flag here:

```go
// +optional
STSEnabled           *bool                    `json:"stsEnabled,omitempty"`
```

Reply:
seems we need have `json:"sts,omitempty"`, as in CRD it's sts